File: ruby/grade-school/grade_school.rb
class School
  attr_reader :database
  VERSION = 1

  def initialize
    @database = {}
  end

  def to_h
    Hash[database.sort].each do |key, value|
      value = value.sort!
    end
  end

  def add(student, grade)
    database[grade] ||= []
    database[grade] << student
  end

  def grade(grade)
    if database[grade]
      database[grade].sort
    else
      []
    end
  end
end

File: javascript/grade-school/grade-school.js
function School() {
  var roster = {}

  this.roster = function() {
    return roster;
  }

  this.add = function(student, grade) {
    if (roster[grade] === undefined ) {
      roster[grade] = [student];
    } else {
      roster[grade].push(student)
      roster[grade] = roster[grade].sort()
    }
  }

  this.grade = function(grade) {
    if (roster[grade]) {
      return roster[grade];
    }
    return [];
  }
}

module.exports = School;
